.github/workflows/deploy-orchard1-to-azure-app-service.yml

```
@@ -11,17 +11,23 @@ on:
   workflow_call:
     secrets:
       CHECKOUT_TOKEN:
+        required: false
         description: >
           The GitHub token to authenticate checkout. Pass in a GitHub personal access token if authenticated submodules
           are used.
+
+      # These secrets are used for Azure authentication through Service Principal Federated Credentials with the
+      # azure/login (https://github.com/azure/login) action, which is proxied by our login-to-azure action below.
       AZURE_APP_SERVICE_DEPLOYMENT_SERVICE_PRINCIPAL_ID:
         required: true
       AZURE_APP_SERVICE_DEPLOYMENT_AZURE_TENANT_ID:
         required: true
       AZURE_APP_SERVICE_DEPLOYMENT_AZURE_SUBSCRIPTION_ID:
         required: true
```

Same for the others:

```suggestion
      AZURE_APP_SERVICE_DEPLOYMENT_SERVICE_PRINCIPAL_ID:
        required: true
        description: >
            The Application (client) ID of the Microsoft Entra ID Service Principal used for the OpenID Connect-based
            authentication with Azure services. Check out the azure/login action's documentation
            (https://github.com/azure/login) on the client-id parameter for details.
      AZURE_APP_SERVICE_DEPLOYMENT_AZURE_TENANT_ID:
        required: true
        description: >
            The tenant ID of the Microsoft Entra ID tenant used for the OpenID Connect-based authentication with Azure
            services. Check out the azure/login action's documentation (https://github.com/azure/login) on the tenant-id
            parameter for details.
      AZURE_APP_SERVICE_DEPLOYMENT_AZURE_SUBSCRIPTION_ID:
        required: true
        description: >
            The subscription ID of the Azure subscription the resources are under. You can look this up e.g. in the
            Azure Portal under any resource or the subscription itself.
```

---

I can't commit this suggestion directly for some reason.